src/ngx_http_lua_module.c

```diff
@@ -205,6 +205,27 @@ static ngx_command_t ngx_http_lua_cmds[] = {
       0,
       (void *) ngx_http_lua_init_worker_by_file },
 
+    { ngx_string("exit_worker_by_lua_block"),
+      NGX_HTTP_MAIN_CONF|NGX_CONF_BLOCK|NGX_CONF_NOARGS,
+      ngx_http_lua_exit_worker_by_lua_block,
+      NGX_HTTP_MAIN_CONF_OFFSET,
+      0,
+      (void *) ngx_http_lua_exit_worker_by_inline },
+
+    { ngx_string("exit_worker_by_lua"),
```

Comment:
I believe you would also need to update `ngx_http_lua_{phase | common | util}.c` to introduce this new phase context? As well as setting the context itself? I could be wrong.

I believe it will also be request of you to introduce tests for this feature.